### PR TITLE
Adding support for transcriptions ( and OpenAI Whisper model)

### PR DESCRIPTION
**Pull request by michaellavelle**

### CODE REVIEW

File: models/spring-ai-openai/src/main/java/org/springframework/ai/openai/OpenAiTranscriptionClient.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+			var transcription = transcriptionEntity.getBody();
+
+			if (transcription == null) {
+				logger.warn("No transcription returned for request: {}", audioResource);
+				return new TranscriptionResponse(null);
+			}
+
+			Transcript transcript = new Transcript(transcription.text());
+
+			RateLimit rateLimits = OpenAiResponseHeaderExtractor.extractAiResponseHeaders(transcriptionEntity);
```
Comment (michaellavelle) [inline models/spring-ai-openai/src/main/java/org/springframework/ai/openai/OpenAiTranscriptionClient.java:92]: I'm re-using these RateLimit and Usage classes from the chat packages here, but given that these are now used outside of chat, should they be moved into a more generic package ?

**Comment by michaellavelle** [comment]
Please see https://github.com/spring-projects/spring-ai/issues/149

File: models/spring-ai-openai/src/main/java/org/springframework/ai/openai/OpenAiTranscriptionClient.java (hunk 1 above)
Comment (markpollack) [reply models/spring-ai-openai/src/main/java/org/springframework/ai/openai/OpenAiTranscriptionClient.java]: I am not yet sure, we need to review this in the 0.9.0 timeline.  I suspect there are cross cutting concerns.  Now we are wrapping up the other direction, input options, to support a tiered structure of portable and model specific options.  The same analysis and likely same approach needs to apply for the metadata info coming out of the response.

**Comment by markpollack** [comment]
Thanks for the contribution, this looks like a very strong PR on first glance!

**Comment by hemeda3** [comment]
Thank you.
Your PR inspired me to do the Text To Audio (Speech API).

[Added support for OpenAI Text to Audio (Speech API )](https://github.com/spring-projects/spring-ai/pull/317/files#top)
#317

**Comment by tzolov** [comment]
Thank you @michaellavelle 
I've replaced the lower-level client and removed the code from the spring-ai-core.  We will need at least few audio and transcription client implementations before we can generalise. 
Reworked, rebased, squashed and merged at 7d04167833a36de3a639d616aa996cde71e29cba

**Pull request closed without merging**